packages/node_modules/@webex/internal-plugin-ediscovery/src/ediscovery.js

```diff
@@ -203,78 +212,43 @@ const EDiscovery = WebexPlugin.extend({
+   * Retrieves a list of the spaces relevant to a specified report
    * @param {UUID} reportId - Id of the report being requested
    * @param {Object} options - optional parameters for this method
-   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 90s
+   * @param {number} options.offset - start position from which to retrieve records
+   * @param {number} options.size - the number of records to retrieve
+   * @param {number} options.timeoutMs - connection timeout in milliseconds, defaults to 30s
    * @returns {Promise<ResponseEntity<ContentSummary>>} Http response containing the content summary
    */
   getContentSummary(reportId, options) {
     if (!reportId) {
       throw Error('Undefined parameter');
     }
 
-    const contentSummaryDefaultOptions = {timeoutMs: 90000};
-
     // use spread operator to set default options
-    options = {...contentSummaryDefaultOptions, ...options};
-
-    let offset = 0;
-    const size = 100;
-    let contentSummary = [];
-
-    function getNextContentSummaryPage(plugin) {
-      return plugin.request({
-        method: 'GET',
-        service: 'ediscovery',
-        resource: `reports/${reportId}/contents/summary`,
-        qs: {offset, size},
-        timeout: options.timeoutMs
-      })
-        .then((res) => {
-          if (!plugin.contentSummaryBySpaceIdCache.has(reportId)) {
-            plugin.contentSummaryBySpaceIdCache.set(reportId, new Map());
-          }
-
-          for (let i = 0; i < res.body.length; i += 1) {
-            if (res.body[i] && res.body[i].spaceId) {
-              try {
-                // The results are written to the contentSummaryBySpaceIdCache since information for
-                // a space is likely to be accessed very frequently when decrypting activities
-                plugin.contentSummaryBySpaceIdCache.get(reportId).set(res.body[i].spaceId, res.body[i]);
-              }
-              catch (error) {
-                plugin.logger.error(`Error adding ${res.body[i].spaceId} to contentSummaryBySpaceIdCache: ${error}`);
-              }
-            }
-            else {
-              plugin.logger.error('Error adding undefined space to contentSummaryBySpaceIdCache');
-            }
-          }
-
-          contentSummary = contentSummary.concat(res.body);
-          if (res.body.length < size) {
-            return contentSummary;
-          }
-          offset += size;
-
-          return getNextContentSummaryPage(plugin);
-        });
-    }
+    options = {...this.config.defaultOptions, ...options};
 
-    return getNextContentSummaryPage(this);
+    return this.request({
+      method: 'GET',
+      service: 'ediscovery',
+      resource: `reports/${reportId}/contents/summary`,
+      qs: {offset: options.offset, size: options.size},
+      timeout: options.timeoutMs
+    })
+      .then((res) => this._writeSpacesToContentSummaryBySpaceIdCache(reportId, res.body)
+        .then(() => Promise.resolve(res)));
```

Comment:
I don't think you need the additional `.then`. If `_writeSpacesToContentSummaryBySpaceIdCache` returns a resolved promise, you can just return it. 

```
.then((res) => this._writeSpacesToContentSummaryBySpaceIdCache(reportId, res.body));
```

Reply:
But since _writeSpacesToContentSummaryBySpaceIdCache doesn't return res, the response won't be passed back to the caller right?

Reply:
Ah, I see. I think the nested `then`s threw me off a bit. I would rewrite this to use `await` to make it clearer. 

```
.then((res) => {
  await this._writeSpacesToContentSummaryBySpaceIdCache(reportId, res.body); 
  return Promise.resolve(res); 
});
```
You're not returning the result of that write to cache, but rather waiting for it to write before simply returning the response.

Reply:
Can await be used from within WebexPlugin?

Reply:
Yes, here is an [example](https://github.com/webex/webex-js-sdk/pull/1304/files#diff-4839efc2cf969a22d9e796384dcf75bdR1141).

Reply:
Thanks Matt, adding async to the enclosing function in the then did the trick